mne/encoding/model.py

```diff
@@ -91,31 +116,37 @@ def fit(self, X, y=None):
         n_features = X.shape[0]
 
         # Fit the model and assign coefficients
-        self.coef_ = np.zeros([n_chs, n_features])
-        for ii in range(n_chs):
-            i_y = y[ii][:, np.newaxis]
-            self.est.fit(X.T, i_y)
-            self.coef_[ii, :] = self.est.steps[-1][-1].coef_
+        self.est.fit(X.T, y.T)
+        self.coef_ = self.est._final_estimator.coef_
         self.X = X
         self.y = y
 
     def predict(self, X, preproc_x=None):
         """Generate predictions using fit coefficients.
 
-        This uses the `coef_` attribute for predictions.
+        This uses the predict method of the final estimator in the
+        `est` attribute.
+
+        Parameters
+        ----------
+        X : array, shape (n_samples, n_features)
+            The input feature array.
         """
+        # Preprocess X with the pipeline
         if preproc_x is not None:
             X = preproc_x.fit_transform(X)
-        X = self.est._pre_transform(X.T)[0].T
-        preds = np.dot(self.coef_, X)
-        return preds
+        return self.est.predict(X.T)
```

Comment:
I wanted to ask you guys about that... right now the MNE object expect the
input to be shape (features, times) to be in accordance with other mne
objects. The transposes are because sklearn wants (times, features). I
think it'd be better to just let our objects be (times, features) so it's
easier to chain and so we avoid confusing transposes. Wdyt?

Sent from my phone, apologies for the curtness and type-os
On Jun 20, 2016 9:37 PM, "Jean-Rémi KING" <notifications@github.com> wrote:

> In mne/encoding/model.py
> <https://github.com/mne-tools/mne-python/pull/3310#discussion_r67806948>:
>
> >          if preproc_x is not None:
> >              X = preproc_x.fit_transform(X)
> > -        X = self.est._pre_transform(X.T)[0].T
> > -        preds = np.dot(self.coef_, X)
> > -        return preds
> > +        return self.est.predict(X.T)
>
> why the transpose here? shouldn't be dealt with the preproc step?
>
> (I m a bit lost sorry)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/mne-tools/mne-python/pull/3310/files/d3b72b2d346c3581140d18ee0f946e266892010f..eef15fb162dc28e6a8feb7216333315b8a6c2ccc#r67806948>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABwSHSRN5xy_LZ07X5czcpdyOHo5B7OPks5qN2p6gaJpZM4I28Ut>
> .
>